### PR TITLE
Submit lesson08 and lesson09

### PR DESCRIPTION
**Pull request by InduKolli**

### CODE REVIEW

File: students/Kollii/lesson08/circle.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    @property
+    def area(self):
+        # Read only attribute
+        return math.pi * (self.radius ** 2)
+
+    @classmethod
+    def from_diameter(cls, diameter):
+        self = cls(diameter/2)
+        return self
```
Comment (natasha-aleksandrova) [inline students/Kollii/lesson08/circle.py]: note that you can just do `return cls(diameter / 2)`

File: students/Kollii/lesson09/mailroom_oo/donor_models.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    def print_thanksnote(self):
+        note = (f'\nDear {self.name},'
+                    f'\nThank you for your generous donation total amount of ${self.donation_amt[-1]:.2f}.'
+                    '\nWe appreciate your support.'
+                    '\n\nRegards,\nXYZ Charity Team\n')
+        return note
+
+    def sort_key(self):
+        return sum(self.donation_amt)
```
Comment (natasha-aleksandrova) [inline students/Kollii/lesson09/mailroom_oo/donor_models.py]: you can also add donation avg/total/sum calculations in this class as well

File: students/Kollii/lesson09/mailroom_oo/donor_models.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    @staticmethod
+    def initialize_dict(donor_data):
+        # create a new collection object
+        donors = DonorCollection()                     
+        for name, amounts in donor_data.items():     
+            donor = Donor(name, amounts)            
+            # add the new donor to the collection
+            donors.add(donor)
+        return donors
```
Comment (natasha-aleksandrova) [inline students/Kollii/lesson09/mailroom_oo/donor_models.py]: nice use of staticmethod here!

File: students/Kollii/lesson09/mailroom_oo/donor_models.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    def add(self, donor):
+        """ Add donor to the collection """
+        self.donors[donor.name] = donor
+
+    def donors_list(self):
+        """ Generate a list of donors """
+        ls=[]
+        for names in self.donors:
+            ls.append(names)
+        return '\n'.join(ls)
```
Comment (natasha-aleksandrova) [inline students/Kollii/lesson09/mailroom_oo/donor_models.py]: you can use a list comprehension here - however, it's even simpler:
`return '\n'.join(list(self.donors))`

**Comment by InduKolli** [comment]
Thank you very much Natasha for the review and comments!
They are very helpful to improve my code and knowledge.

Regards,

Indu
________________________________

**Comment by natasha-aleksandrova** [comment]
my pleasure! good luck in the next course :)

**Comment by InduKolli** [comment]
Thank you so much Natasha!
Do I get any email or notification that I finished this course?
Please let me know.

Regards,

Indu


________________________________

**Comment by natasha-aleksandrova** [comment]
yes you will - you can email us to get more information on the process.

**Comment by InduKolli** [comment]
Thank you Natasha!
I will certainly let you know if I need information.

Thank you again for all your support and help!

Indu
________________________________

**Pull request merged**